### ISSUE TITLE
./tests/TESTrun gives '0 tests failed' even when there are failed tests

### ISSUE THREAD

**Issue by fxlb** [the-tcpdump-group/tcpdump#824]
Debian bullseye (testing)
```
./configure --with-system-libpcap; make -s clean all; ./tcpdump --version
tcpdump version 4.10.0-PRE-GIT
libpcap version 1.9.1 (with TPACKET_V3)
OpenSSL 1.1.1d  10 Sep 2019
SMI-library: 0.4.8
```
```
./tests/TESTrun
Running tests from /home/cpe/soft-origin/tcpdump/my-tcpdump/tests
    print-x                                 : passed
[...]
    bgp-encap                               : passed
    brcmtag                                 : TEST FAILED(exit core=0/diffstat=1,0/r=256) with error messages:
reading from file brcm-tag.pcap, link-type 281, snapshot length 262144
tcpdump: unknown data link type 281
    brcmtag-e                               : TEST FAILED(exit core=0/diffstat=1,0/r=256) with error messages:
reading from file brcm-tag.pcap, link-type 281, snapshot length 262144
tcpdump: unknown data link type 281
    brcmtagprepend                          : TEST FAILED(exit core=0/diffstat=1,0/r=256) with error messages:
reading from file brcm-tag-prepend.pcap, link-type 282, snapshot length 262144
tcpdump: unknown data link type 282
    dsa                                     : TEST FAILED(exit core=0/diffstat=1,0/r=256) with error messages:
reading from file dsa.pcap, link-type 284, snapshot length 262144
tcpdump: unknown data link type 284
    dsa-e                                   : TEST FAILED(exit core=0/diffstat=1,0/r=256) with error messages:
reading from file dsa.pcap, link-type 284, snapshot length 262144
tcpdump: unknown data link type 284
    eapon1                                  : passed
    edsa                                    : TEST FAILED(exit core=0/diffstat=1,0/r=256) with error messages:
reading from file edsa.pcap, link-type 285, snapshot length 262144
tcpdump: unknown data link type 285
    edsa-e                                  : TEST FAILED(exit core=0/diffstat=1,0/r=256) with error messages:
reading from file edsa.pcap, link-type 285, snapshot length 262144
tcpdump: unknown data link type 285
    esp0                                    : passed
[...]
------------------------------------------------
   0 tests failed
 516 tests passed

Failed test: brcmtag

1,23c1
<     1  02:10:20.634030 BRCM tag OP: IG, TC: 0, TE: None, TS: 0, DST map: 0x0080, IP 0.0.0.0.68 > 255.255.255.255.67: BOOTP/DHCP, Request from 00:10:18:de:38:1e, length 300
<     2  02:10:20.696008 BRCM tag OP: IG, TC: 0, TE: None, TS: 0, DST map: 0x0020, IP 0.0.0.0.68 > 255.255.255.255.67: BOOTP/DHCP, Request from 00:10:18:de:38:1e, length 300
[...]
```

**Comment by infrastation** [comment]
Does this still stand?

**Comment by fxlb** [comment]
Yes.

**Comment by fxlb** [comment]
7 tests failed but display:
"0 tests failed"

**Comment by fxlb** [comment]
Fixed now. Closing.

**Comment by guyharris** [comment]
Probably fixed by 701100567bf15763e33d9ce281b9cb49cbc8e83a.
